src/Symfony/Component/Routing/Generator/Dumper/PhpGeneratorDumper.php

```diff
@@ -105,10 +106,21 @@ private function generateGenerateMethod()
         return <<<'EOF'
     public function generate($name, $parameters = array(), $referenceType = self::ABSOLUTE_PATH)
     {
+        $locale = $parameters['_locale']
+            ?? $this->context->getParameter('_locale')
+            ?: $this->defaultLocale;
+
+        if (isset(self::$declaredRoutes[$name.'.'.$locale])) {
+            unset($parameters['_locale']);
+            $name = $name.'.'.$locale;
+            goto generate;
```

Comment:
instead of the goto, the next "if" could just be an "elseif", isn't it?

Reply:
If I'd use an elseif then I would still need to use another else to throw the exception. It's all the same to me in the end, I thought this would create an easier to understand flow, but if it doesn't do that I'll happily revert it.

Reply:
An "else" would be less surprising for the future reader so let's change yes please :)